cloudsql-mysql-plugin/src/e2e-test/java/io/cdap/plugin/CloudMySqlClient.java

```diff
@@ -0,0 +1,159 @@
+    ResultSetMetaData mdSource = rsSource.getMetaData();
+    ResultSetMetaData mdTarget = rsTarget.getMetaData();
+    int columnCountSource = mdSource.getColumnCount();
+    int columnCountTarget = mdTarget.getColumnCount();
+    Assert.assertEquals("Number of columns in source and target are not equal", columnCountSource, columnCountTarget);
+    while (rsSource.next() && rsTarget.next()) {
+      int currentColumnCount = 1;
+      while (currentColumnCount <= columnCountSource) {
+        String columnTypeName = mdSource.getColumnTypeName(currentColumnCount);
+        int columnType = mdSource.getColumnType(currentColumnCount);
+        String columnName = mdSource.getColumnName(currentColumnCount);
+        if (columnType == Types.TIMESTAMP) {
+          GregorianCalendar gc = new GregorianCalendar(TimeZone.getTimeZone("UTC"));
+          gc.setGregorianChange(new Date(Long.MIN_VALUE));
+          Timestamp sourceTS = rsSource.getTimestamp(currentColumnCount, gc);
+          Timestamp targetTS = rsTarget.getTimestamp(currentColumnCount, gc);
+          Assert.assertTrue(String.format("Different values found for column : %s", columnName),
+                            sourceTS.equals(targetTS));
+        } else {
+          String sourceString = rsSource.getString(currentColumnCount);
+          String targetString = rsTarget.getString(currentColumnCount);
+          Assert.assertTrue(String.format("Different values found for column : %s", columnName),
+                            String.valueOf(sourceString).equals(String.valueOf(targetString)));
+        }
+        currentColumnCount++;
+      }
+    }
+    Assert.assertFalse("Number of rows in Source table is greater than the number of rows in Target table",
+                       rsSource.next());
+    Assert.assertFalse("Number of rows in Target table is greater than the number of rows in Source table",
+                       rsTarget.next());
+    return true;
+  }
+
+  public static void createSourceTable(String sourceTable) throws SQLException, ClassNotFoundException {
+    try (Connection connect = getCloudSqlConnection(); Statement statement = connect.createStatement()) {
+      String createSourceTableQuery = "CREATE TABLE IF NOT EXISTS " + sourceTable + "(id int, lastName varchar(255), PRIMARY KEY (id))";
+      statement.executeUpdate(createSourceTableQuery);
+
+      // Truncate table to clean the data of last failure run.
+      String truncateSourceTableQuery = "TRUNCATE TABLE " + sourceTable;
+      statement.executeUpdate(truncateSourceTableQuery);
+
+      // Insert dummy data.
+      statement.executeUpdate("INSERT INTO " + sourceTable + " (id, lastName)" + "VALUES (1, 'Priya')");
+      statement.executeUpdate("INSERT INTO " + sourceTable + " (id, lastName)" + "VALUES (2, 'Shubhangi')");
+      statement.executeUpdate("INSERT INTO " + sourceTable + " (id, lastName)" + "VALUES (3, 'Shorya')");
```

Comment:
@priyabhatnagar25 @IshaCS This PR was supposed to be raised externally are we planning to raise this ?